datalad/distribution/get.py

```diff
@@ -262,6 +270,9 @@ def __call__(
                         relpath(opj(ds_path, local_results[i]['file']), ds.path)
 
             global_results.extend(local_results)
+
+        if not found_an_annex:
+            lgr.warning("Found no annex. Could not perform any get operation.")
```

Comment:
I am ok with such approach instead of mine... just fix up the test... I forgot also to improve that test by making an annex subdataset and testing actual recursive call on super pure-git dataset... we really need those tests